### PR TITLE
[CI] Switch to macos-latest

### PR DESCRIPTION
**Pull request by srittau**
macos-12 is deprecated and will be removed:
https://github.com/actions/runner-images/issues/10721

### CODE REVIEW

File: stdlib/@tests/stubtest_allowlists/darwin-py38.txt [hunk 1]
```diff
@@ -5,3 +5,10 @@ spwd
 _msi
 msilib(.[a-z]+)?
 ossaudiodev
+
+# Added in Python 3.8.14
+def set_int_max_str_digits(maxdigits: int) -> None: ...
+def get_int_max_str_digits() -> int: ...
```
Comment (AlexWaygood) [inline stdlib/@tests/stubtest_allowlists/darwin-py38.txt:11]:
```suggestion
sys.set_int_max_str_digits
sys.get_int_max_str_digits
```

File: stdlib/@tests/stubtest_allowlists/darwin-py39.txt [hunk 2]
```diff
@@ -5,3 +5,10 @@ spwd
 _msi
 msilib(.[a-z]+)?
 ossaudiodev
+
+# Added in Python 3.9.14
+def set_int_max_str_digits(maxdigits: int) -> None: ...
+def get_int_max_str_digits() -> int: ...
```
Comment (AlexWaygood) [inline stdlib/@tests/stubtest_allowlists/darwin-py39.txt:11]:
```suggestion
sys.set_int_max_str_digits
sys.get_int_max_str_digits
```

File: stdlib/@tests/stubtest_allowlists/darwin-py38.txt (hunk 1 above)
Comment (srittau) [reply stdlib/@tests/stubtest_allowlists/darwin-py38.txt]: Nothing to see here ...

**Review by AlexWaygood — APPROVED** [review]

**Review by hauntsaninja — APPROVED** [review]
Thanks!

**Pull request merged**